packages/netlify-cms-core/src/lib/registry.js

```diff
@@ -78,25 +80,32 @@ export function registerWidget(name, control, preview) {
       name: widgetName,
       controlComponent: control,
       previewComponent: preview,
+      allowMapValue,
       globalStyles,
+      ...options
     } = name;
     if (registry.widgets[widgetName]) {
-      console.error(oneLine`
+      console.warn(oneLine`
         Multiple widgets registered with name "${widgetName}". Only the last widget registered with
         this name will be used.
       `);
     }
     if (!control) {
       throw Error(`Widget "${widgetName}" registered without \`controlComponent\`.`);
     }
-    registry.widgets[widgetName] = { control, preview, globalStyles };
+    registry.widgets[widgetName] = { control, preview, globalStyles, allowMapValue, ...options };
   } else {
     console.error('`registerWidget` failed, called with incorrect arguments.');
   }
 }
 export function getWidget(name) {
   return registry.widgets[name];
 }
+export function getWidgets() {
+  return produce(Object.entries(registry.widgets), draft => {
+    return draft.map(([key, value]) => ({ name: key, ...value }));
+  });
+}
```

Comment:
Method for grabbing all widgets without passing the actual object. Took the opportunity to introduce immer as a portable immutability provider.